micropython/umqtt.simple/umqtt/simple.py

```diff
@@ -16,8 +16,7 @@ def __init__(
         user=None,
         password=None,
         keepalive=0,
-        ssl=False,
-        ssl_params={},
```

Comment:
This seems now like a "breaking" change, one needs to create and pass a `SSLContext`, see e.g. https://github.com/orgs/micropython/discussions/13624, I think this should be something like 
```py

if self.ssl:
             if hasattr(self.ssl, "wrap_socket"):
                 self.sock = self.ssl.wrap_socket(self.sock, **self.ssl_params)
             else:
                 import ssl as _ssl

                 self.sock = _ssl.wrap_socket(self.sock, **self.ssl_params)
```

Reply:
Yes, I was not aware that breaking changes here are not possible.  I would argue that this breaking change is good here.

Reply:
I think this breaking change is a fair change to make.  We need to improve and make progress on things and sometimes it's not practical to retain backwards compatibility on everything.

In this case using the new library with old user code will raise an exception if the `ssl` argument is used.  So it's easy for the user to know that things need to be changed.

And the new scheme matches better how SSL works in `asyncio`, passing in an `SSLContext`.

In the end a user can just keep using an old version of `umqtt` if needed.

Reply:
Then document this somewhere in the `umqtt`'s README or next release info should be fine 👍🏼